waku/waku_archive/driver/postgres_driver/postgres_driver.nim

```diff
@@ -29,93 +29,101 @@ type PostgresDriver* = ref object of ArchiveDriver
 
 const SelectWithCursorAscStmtName = "SelectWithCursorAsc"
 const SelectWithCursorAscStmtDef =
-  """SELECT storedAt, contentTopic, payload, pubsubTopic, version, timestamp, id, messageHash, meta FROM messages
+  SelectClause &
+  """WHERE contentTopic IN ($1) AND
+          messageHash IN ($2) AND
+          pubsubTopic = $3 AND
+          (timestamp, messageHash) > ($4,$5) AND
+          timestamp >= $6 AND
+          timestamp <= $7
+    ORDER BY timestamp ASC, messageHash ASC LIMIT $8;"""
+
+const SelectWithCursorNoDataAscStmtName = "SelectWithCursorNoDataAsc"
+const SelectWithCursorNoDataAscStmtDef =
+  """SELECT messageHash FROM messages
     WHERE contentTopic IN ($1) AND
           messageHash IN ($2) AND
           pubsubTopic = $3 AND
-          (storedAt, messageHash) > ($4,$5) AND
-          storedAt >= $6 AND
-          storedAt <= $7
-    ORDER BY storedAt ASC, messageHash ASC LIMIT $8;"""
+          (timestamp, messageHash) > ($4,$5) AND
+          timestamp >= $6 AND
+          timestamp <= $7
+    ORDER BY timestamp ASC, messageHash ASC LIMIT $8;"""
 
 const SelectMessageByHashName = "SelectMessageByHash"
 const SelectMessageByHashDef =
-  """SELECT storedAt, contentTopic, payload, pubsubTopic, version, timestamp, id, messageHash, meta FROM messages WHERE messageHash = $1"""
-
-const SelectNoCursorV2AscStmtName = "SelectWithoutCursorV2Asc"
-const SelectNoCursorV2AscStmtDef =
-  """SELECT storedAt, contentTopic, payload, pubsubTopic, version, timestamp, id, messageHash, meta FROM messages
-    WHERE contentTopic IN ($1) AND
-          pubsubTopic = $2 AND
-          storedAt >= $3 AND
-          storedAt <= $4
-    ORDER BY storedAt ASC LIMIT $5;"""
-
-const SelectNoCursorV2DescStmtName = "SelectWithoutCursorV2Desc"
-const SelectNoCursorV2DescStmtDef =
-  """SELECT storedAt, contentTopic, payload, pubsubTopic, version, timestamp, id, messageHash, meta FROM messages
-    WHERE contentTopic IN ($1) AND
-          pubsubTopic = $2 AND
-          storedAt >= $3 AND
-          storedAt <= $4
-    ORDER BY storedAt DESC LIMIT $5;"""
-
-const SelectWithCursorV2DescStmtName = "SelectWithCursorV2Desc"
-const SelectWithCursorV2DescStmtDef =
-  """SELECT storedAt, contentTopic, payload, pubsubTopic, version, timestamp, id, messageHash, meta FROM messages
-    WHERE contentTopic IN ($1) AND
-          pubsubTopic = $2 AND
-          (storedAt, id) < ($3,$4) AND
-          storedAt >= $5 AND
-          storedAt <= $6
-    ORDER BY storedAt DESC LIMIT $7;"""
-
-const SelectWithCursorV2AscStmtName = "SelectWithCursorV2Asc"
-const SelectWithCursorV2AscStmtDef =
-  """SELECT storedAt, contentTopic, payload, pubsubTopic, version, timestamp, id, messageHash, meta FROM messages
-    WHERE contentTopic IN ($1) AND
-          pubsubTopic = $2 AND
-          (storedAt, id) > ($3,$4) AND
-          storedAt >= $5 AND
-          storedAt <= $6
-    ORDER BY storedAt ASC LIMIT $7;"""
+  """SELECT timestamp FROM messages
```

Comment:
When selecting a message by hash, why do we only take the timestamp?

Reply:
> Maybe it should be named SelectCursor or something?

I think so, it would be better! The naming `SelectMessage` is kind of misleading in this case I think